internal/secrets/aws.go

```diff
@@ -0,0 +1,56 @@
+package secrets
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+)
+
+type awsSecretsManager struct {
+	client *secretsmanager.SecretsManager
+	logger *slog.Logger
+}
+
+func newAWSSecretsManager(_ context.Context, logger *slog.Logger, url *url.URL) (secretAPI, error) {
+	s, err := session.NewSession(&aws.Config{
+		Region: getRegion(url.Host),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create secret manager session: %w", err)
+	}
+
+	return &awsSecretsManager{
+		client: secretsmanager.New(s, aws.NewConfig()),
+		logger: logger,
+	}, nil
+}
+
+func (a *awsSecretsManager) getSecretValue(key string) (string, bool) {
```

Comment:
We can probably pass context and use the client's `GetSecretValueWithContext()` method. 
https://docs.aws.amazon.com/sdk-for-go/api/service/secretsmanager/#SecretsManager.GetSecretValueWithContext